src/main/java/com/gtnewhorizons/modularui/common/widget/SlotWidget.java

```diff
@@ -606,19 +603,38 @@ && getScreen().inventorySlots.canDragIntoSlot(slotIn)) {
     }
 
     /**
-     * Adapted from {@link LayoutManager#renderSlotOverlay}
+     * Adapted from {@link GuiContainerManager#renderSlotUnderlay}
      */
-    protected void renderSlotOverlayNEI() {
-        ItemStack item = slot.getStack();
-        if (SearchField.searchInventories() && (item == null ? !getSearchExpression().equals("")
-                : !LayoutManager.searchField.getFilter().matches(item))) {
-            GL11.glDisable(GL11.GL_LIGHTING);
-            GL11.glDisable(GL11.GL_DEPTH_TEST);
-            GL11.glTranslatef(0, 0, 150);
-            drawRect(0, 0, 16, 16, 0x80000000);
-            GL11.glTranslatef(0, 0, -150);
-            GL11.glEnable(GL11.GL_LIGHTING);
-            GL11.glEnable(GL11.GL_DEPTH_TEST);
+    protected void renderSlotUnderlayNEI(Slot slotIn) {
+        final int xDisplayPosition = slotIn.xDisplayPosition;
+        final int yDisplayPosition = slotIn.yDisplayPosition;
+        final ModularGui screen = getScreen();
+        slotIn.xDisplayPosition = 1;
+        slotIn.yDisplayPosition = 1;
+
+        for (IContainerDrawHandler drawHandler : GuiContainerManager.drawHandlers) {
+            drawHandler.renderSlotUnderlay(screen, slotIn);
+        }
```

Comment:
I think `GuiContainerManager.getManager().renderSlotUnderlay()` can be used instead of manually looping?